platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -258,6 +258,25 @@ - (id)mgl_jsonExpressionObject {
     return [self valueForKeyPath:@"mgl_jsonExpressionObject"];
 }
 
+- (id)mgl_case {
```

Comment:
This implementation essentially has the developer writing something like this:

```objc
[NSExpression expressionWithFormat:@"FUNCTION({firstConditional, firstExpression, secondConditional, secondExpression, …, falseExpression}, 'mgl_case')"]
```

It’s good that the arguments all sit side by side now instead of requiring nested `TERNARY()` expressions. However, placing all the arguments inside a single aggregate expression would be inappropriate, because `case` behaves like a control structure rather than a typical mathematical function.

Per https://github.com/mapbox/mapbox-gl-native/pull/11450#discussion_r174859821, I think this function should be used more like this:

```objc
[NSExpression expressionWithFormat:@"FUNCTION(firstConditional, 'mgl_case:', firstValue, secondConditional, secondValue, …, falseValue)"]
```

It looks weird, but we need an operand, and it’s natural for the first conditional to be that operand. This design would require a category method of NSPredicate with the following vararg signature:

```objc
- (id)mgl_case:(id)firstValue, ...;
```

As part of #11472, we can implement a control structure–like `MGL_SWITCH:` function that’ll have a call site like this:

```objc
[NSExpression expressionWithFormat:@"MGL_SWITCH(firstConditional, firstValue, secondConditional, secondValue, ..., falseValue)"]
```